coffee-backend/src/main/kotlin/chat/letscoffee/friends/FriendRequestService.kt

```diff
@@ -0,0 +1,44 @@
+package chat.letscoffee.friends
+
+import chat.letscoffee.exception.BadRequestException
+import chat.letscoffee.exception.ResourceNotFoundException
+import chat.letscoffee.security.model.User
+import chat.letscoffee.security.repository.UserRepository
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+
+@Service
+class FriendRequestService(private val repository: FriendRequestRepository, private val userRepository: UserRepository) {
+
+    fun addFriendRequest(from: User, to: String): FriendRequestModel {
+        val touser: User = userRepository.findByNameOrEmail(to, to)?: throw ResourceNotFoundException("User", "username", to);
```

Comment:
`name` is not unique, so it is not a safe way to find a user. Use either `email`